### PR TITLE
element-desktop: remove gcc references

### PR DESCRIPTION
**Pull request by elohmeier**
###### Motivation for this change
I've noticed element-desktop native modules pull in unneeded gcc dependencies.

###### Things done
Removed the gcc dependencies reducing the closure size. Ran nixpkgs-fmt.

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sumnerevans** [comment]
Does this affect seshat's functionality at all?

File: pkgs/applications/networking/instant-messengers/element/keytar/default.nix [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
+, fixup_yarn_lock
+, yarn
+, pkg-config
+, libsecret
+, xcbuild
+, Security
+, AppKit
+, fetchYarnDeps
+, removeReferencesTo
+}:
```
Comment (Ma27) [inline pkgs/applications/networking/instant-messengers/element/keytar/default.nix:16]: This doesn't seem related to the change. *If* we do this formatting, we should at least do it in a second commit.

**Comment by symphorien** [comment]
you could add stdenv.cc.cc to disallowedReferences to prevent unintentional regressions. https://nixos.org/manual/nix/stable/expressions/advanced-attributes.html?highlight=disallowed#advanced-attributes

File: pkgs/applications/networking/instant-messengers/element/keytar/default.nix (hunk 1 above)
Comment (elohmeier) [reply pkgs/applications/networking/instant-messengers/element/keytar/default.nix]: Thanks, I've undone that.

**Comment by elohmeier** [comment]
Thanks for your feedback, I've updated the PR accordingly and rebased onto master.

**Comment by symphorien** [comment]
I tested that element desktop encrypted search still works.
